src/optimise1.jl

```diff
@@ -141,17 +138,30 @@ function minimise_icp(f::Function, x::IntervalBox{N, T}; reltol=1e-3, abstol=1e-
     return lb..global_minimum, arg_minima
 end
 
-function minimise_icp_constrained(f::Function, x::IntervalBox{N, T}, constraints::Vector{Constraint{T}} = Vector{Constraint{T}}(); reltol=1e-3, abstol=1e-3) where {N, T<:Real}
+
+struct ConstraintCond{T}
+    f ::Operation
+    c ::Interval{T}
+end
+
+
+function minimise_icp_constrained(f::Function, x::IntervalBox{N,T}, var, constraints::Vector{ConstraintCond{T}} = Vector{ConstraintCond{T}}(); reltol=1e-3, abstol=1e-3) where {N, T<:Real}
 
     Q = binary_minheap(IntervalBoxMinimum{N, T})
 
     global_minimum = ∞
 
     for t in constraints
-        x = icp(t.f, x, t.c)
+        C = Contractor(t.f, var)
```

Comment:
It is expensive to create a `Contractor`, so this should be done only once for each constraint.

Reply:
Oh right. We should built contractor for each constraint condition only once.